core/commands/files/files.go

```diff
@@ -471,7 +473,12 @@ Example:
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
-		defer fi.Close()
+
+		if noflush {
+			defer fi.Sync()
+		} else {
+			defer fi.Close()
+		}
```

Comment:
```sh
ipfs files write --no-flush $args...
kill -9 $ipfs_pid
ipfs files write --no-flush $args...
```
this
- MUST be _safe_ and not leave the dag in an unrecoverable state.
- SHOULD be _consistent_ and keep the result of the first write (journaled)